core/src/data/function.rs

```diff
@@ -0,0 +1,28 @@
+use {
+    crate::ast::{Expr, OperateFunctionArg},
+    serde::{Deserialize, Serialize},
+};
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct CustomFunction {
+    pub func_name: String,
+    pub args: Option<Vec<OperateFunctionArg>>,
+    pub return_: Option<Expr>,
```

Comment:
1. how about `body` rather than `return_`?
2. is that any use case of using `None` for `return_`?
I'm not sure about supporting custom function without function body expr.
if it doesn't, then we can filter out `None` case and remove `Option`.